pydantic/_internal/_model_construction.py

```diff
@@ -138,6 +156,16 @@ def wrapped_model_post_init(self: BaseModel, context: Any, /) -> None:
 
             cls: type[BaseModel] = super().__new__(mcs, cls_name, bases, namespace, **kwargs)  # type: ignore
 
+            # check that this isn't RootModel
+            if '__pydantic_root_model__' not in namespace:
+                for method_name in pydantic_methods:
+                    if method_name in namespace:
+                        warnings.warn(
+                            f'`{method_name}` is a pydantic method. Overriding this method in `{cls.__name__}` may lead to unexpected behavior.',
+                            category=UserWarning,
+                            stacklevel=2,
+                        )
```

Comment:
I'm not sure having this warning is a good idea. Overriding methods from parent classes is a common practice, especially because we recommended users in the past to do so if they want to modify some default values of arguments.

My understanding was that we were going to warn only if methods are overridden by fields/vice-versa?

Reply:
I'm not even opposed to skipping warnings entirely other than for the protected namespace...

Reply:
Curious to see what @samuelcolvin thinks here...